### PR TITLE
Blaze: abort early if JP not connected and defensive coding re get_current_screen

### PR DESCRIPTION
**Pull request by kraftbj**
Fixes fatal issue on https://wordpress.org/gutenberg for Blaze assuming the editors will always be in the admin context.

## Proposed changes:

* Move should_initialize earlier (the site impacted is not connected to Jetpack, simply installed by way of multisite), so Blaze was causing a fatal to the frontend of a disconnected site.
* Defensive check for get_current_screen. I don't know what we should expect for a frontend Gutenberg in terms of Blaze, but better to not fatal and miss Blaze on the edge case for now.

### Other information:

- [ ] Have you written new tests for your changes, if applicable?
- [ ] Have you checked the E2E test CI results, and verified that your changes do not break them?
- [ ] Have you tested your changes on WordPress.com, if applicable (if so, you'll see a generated comment below with a script to run)?

## Jetpack product discussion

p1676401831382259-slack-C01U2KGS2PQ
## Does this pull request change what data or activity we track or use?

No.

## Testing instructions:

* Put Gutenberg on a page outside of wp-admin (use this theme: p1676403580324799-slack-C01U2KGS2PQ )
* Have an unconnected installed Jetpack
* Before patch: see fatal.
* After patch: no fatal

### CODE REVIEW

File: projects/packages/blaze/src/class-blaze.php [hunk 1]
```diff
@@ -214,12 +214,20 @@ public static function jetpack_blaze_row_action( $post_actions, $post ) {
 	 * Enqueue block editor assets.
 	 */
 	public static function enqueue_block_editor_assets() {
+		// Bail if criteria is not met to enable Blaze features.
+		if ( ! self::should_initialize() ) {
+			return;
+		}
```
Comment (anomiex) [inline projects/packages/blaze/src/class-blaze.php]: Personally, I'd leave this where it was. The check for the current screen looks cheaper than the connection check, so better to do that first. They're still both being done before any other checks.

Comment (kraftbj) [reply projects/packages/blaze/src/class-blaze.php]: that's fair. I'll switch them back around. Having the `get_current_screen` first will prevent the fatal, which is the ultimate initial goal. it fataling for a disconnected site got under my skin, but you're right that if the fatal is covered, it doesn't matter.

Comment (kraftbj) [reply projects/packages/blaze/src/class-blaze.php]: [f3a8613](https://github.com/Automattic/jetpack/pull/28955/commits/f3a8613462181292a96fcbcd1cf527f24b81f7de)

File: projects/packages/blaze/src/class-blaze.php [hunk 2]
```diff
@@ -215,11 +215,14 @@ public static function jetpack_blaze_row_action( $post_actions, $post ) {
...
-		 * We do not want (nor need) Blaze in the site editor, or the widget editor, or the classic editor.
-		 * We only want it in the post editor.
-		 * Enqueueing the script in those editors would cause a fatal error.
-		 * See #20357 for more info.
-		 */
+		* We do not want (nor need) Blaze in the site editor, or the widget editor, or the classic editor.
+		* We only want it in the post editor.
+		* Enqueueing the script in those editors would cause a fatal error.
+		* See #20357 for more info.
+		*/
```
Comment (anomiex) [inline projects/packages/blaze/src/class-blaze.php]: The indenting here somehow got screwed up
```suggestion
		 * We do not want (nor need) Blaze in the site editor, or the widget editor, or the classic editor.
		 * We only want it in the post editor.
		 * Enqueueing the script in those editors would cause a fatal error.
		 * See #20357 for more info.
		 */
```

Comment (kraftbj) [reply projects/packages/blaze/src/class-blaze.php]: PHPCBF fail. Thanks!

**Review by jeherve — APPROVED** [review]

**Comment by samiff** [comment]
https://packagist.org/packages/automattic/jetpack-blaze#v0.5.1.2

Added to Jetpack plugin v11.8.4
